MaterialLib/MPL/Properties/PermeabilityMohrCoulombFailureIndexModel.cpp

```diff
@@ -0,0 +1,156 @@
+
+#include <boost/math/constants/constants.hpp>
+#include <cmath>
+#include <limits>
+
+#include "BaseLib/Error.h"
+#include "MaterialLib/MPL/Medium.h"
+#include "MaterialLib/MPL/Utils/FormEigenTensor.h"
+#include "MaterialLib/MPL/Utils/GetSymmetricTensor.h"
+#include "MathLib/KelvinVector.h"
+#include "MathLib/MathTools.h"
+#include "ParameterLib/CoordinateSystem.h"
+#include "ParameterLib/Parameter.h"
+
+namespace MaterialPropertyLib
+{
+template <int DisplacementDim>
+PermeabilityMohrCoulombFailureIndexModel<DisplacementDim>::
+    PermeabilityMohrCoulombFailureIndexModel(
+        std::string name, ParameterLib::Parameter<double> const& k0,
+        double const kr, double const b, double const c, double const phi,
+        double const k_max, double const t_sigma_max,
+        ParameterLib::CoordinateSystem const* const local_coordinate_system)
+    : k0_(k0),
+      kr_(kr),
+      b_(b),
+      c_(c),
+      phi_(boost::math::constants::degree<double>() * phi),
+      k_max_(k_max),
+      t_sigma_max_(t_sigma_max),
+      local_coordinate_system_(local_coordinate_system)
+{
+    name_ = std::move(name);
+}
+
+template <int DisplacementDim>
+void PermeabilityMohrCoulombFailureIndexModel<DisplacementDim>::checkScale()
+    const
+{
+    if (!std::holds_alternative<Medium*>(scale_))
+    {
+        OGS_FATAL(
+            "The property 'PermeabilityMohrCoulombFailureIndexModel' is "
+            "implemented on the 'medium' scale only.");
+    }
+}
+
+template <int DisplacementDim>
+PropertyDataType
+PermeabilityMohrCoulombFailureIndexModel<DisplacementDim>::value(
+    VariableArray const& variable_array,
+    ParameterLib::SpatialPosition const& pos, double const t,
+    double const /*dt*/) const
+{
+    auto const& stress_vector = std::get<SymmetricTensor<DisplacementDim>>(
+        variable_array[static_cast<int>(Variable::stress)]);
+
+    auto const& stress_tensor =
+        formEigenTensor<3>(static_cast<PropertyDataType>(stress_vector));
+
+    Eigen::SelfAdjointEigenSolver<Eigen::Matrix<double, 3, 3>>
+        eigenvalue_solver(stress_tensor);
+
+    // Principle stress
+    auto const sigma = eigenvalue_solver.eigenvalues();
+
+    auto k_data = k0_(t, pos);
+
+    double const max_sigma = std::max(std::fabs(sigma[0]), std::fabs(sigma[2]));
+
+    if (max_sigma < std::numeric_limits<double>::epsilon())
+    {
+        return fromVector(k_data);
+    }
+
+    double const sigma_m = 0.5 * (sigma[2] + sigma[0]);
+    double f = 0.0;
+    if (sigma_m > 0.5 * c_ / std::tan(phi_))
+    {
+        // von Mises failure criterion
+        double val = 0.0;
+        for (int i = 0; i < 3; i++)
+        {
+            int const i1 = i % 3;
+            int const i2 = (i + 1) % 3;
+            val += (sigma[i2] - sigma[i1]) * (sigma[i2] - sigma[i1]);
+        }
+        double J = std::sqrt(0.5 * val);
+        f = J / t_sigma_max_;
+    }
+    else
+    {
+        // von Mohr Coulomb failure criterion
+        // +- tau_t
```

Comment:
What is tau_t? It is not mentioned in the documentation, which has tau and tau_m.

Reply:
Dropped that line.